miner/worker.go

```diff
@@ -516,7 +547,7 @@ func (self *worker) commitNewWork() {
 		glog.V(logger.Info).Infof("commit new work on block %v with %d txs & %d uncles. Took %v\n", work.Block.Number(), work.tcount, len(uncles), time.Since(tstart))
 		self.unconfirmed.Shift(work.Block.NumberU64() - 1)
 	}
-	self.push(work)
+	go self.push(work)
```

Comment:
**Note to self: investigate this! The `go` spawning is needed due to our event system blocking up on delivering chain head events for multiple subscriptions in the same goroutine (work listening + miner strat). This is the only logic change to the miner while introducing the strategy, and I'd rather work around having to change this (simply because I don't want to break anything).**